src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,10 +3,22 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')">
-    <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
-    <LangVersion Condition="'$(LangVersion)' == ''">$(MaxSupportedLangVersion)</LangVersion>
+  <PropertyGroup>
+    <!-- .NETCoreApp < 3.0, .NETStandard < 2.1, or any other target framework -->
+    <MaxSupportedLangVersion Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
```

Comment:
Consider making this an `_` prefixed property to make it clear that it's a property we don't want users setting themselves.

Reply:
We don't actually need MaxSupportedLangVersion if we're not letting users set it, but i'm going to keep with an `_` prefix to aid potential debugging of msbuild logs.

---
In reply to: [440427516](https://github.com/dotnet/roslyn/pull/44911#discussion_r440427516) [](ancestors = 440427516)

Reply:
We were setting `$(MaxSupportedLangVersion)` for Xamarin projects, to opt into C# 8.0 by default.

Should we use `$(_MaxSupportedLangVersion)` now instead? (even though it is considered private)

Context: https://github.com/xamarin/xamarin-android/issues/5049#issuecomment-680950804

Reply:
Why aren't you all using the literal `Latest`? It was meant for exactly this purpose: represent the latest RTM version of the language.

Reply:
It might not be a concern in practice, but in theory language features that require runtime support might not yet be implemented in the Mono runtime.

Reply:
Generally speaking, those features gracefully degrade and issue warnings when used on unsupported runtimes. For example, if you attempt to use a platform-default unmanaged function pointer on a runtime that does not support it (does not include [this](https://source.dot.net/#System.Private.CoreLib/RuntimeFeature.cs,9ae2f99bc8abeb00,references) api), the compiler will issue an error to state that it's not supported.

Reply:
To be clear, the Xamarin MSBuild targets should include this by default instead?
```xml
<LangVersion Condition="'$(LangVersion)' == ''">latest</LangVersion>
```

Reply:
> It might not be a concern in practice, but in theory language features that require runtime support might not yet be implemented in the Mono runtime.

We don't flip to latest being the new C# version until basically right before we release. The flip to latest being 9 happened just a week a go. So this seems pretty unlikely given htat we coordinate the runtime changes.

Reply:
This prompted some good discussion that we're working through.  The Mono runtime changes are done in `master`, on the "netcore" version, originally intended for `net5`, but bumped back to `net6`.  This isn't the version that android and iOS use *today*.  So we will need to look at backporting those changes to the Mono branch we currently use, or figure out what our C#9 support story is.